include/hal/library/cryptlib.h

```diff
@@ -1740,6 +1740,25 @@ bool libspdm_x509_get_issuer_name(const uint8_t *cert, size_t cert_size,
                                   uint8_t *cert_issuer,
                                   size_t *issuer_size);
 
+/**
+ * Retrieve the public key encrypt algo oid from one X.509 certificate.
+ *
+ * @param[in]      cert                 Pointer to the DER-encoded X509 certificate.
+ * @param[in]      cert_size            size of the X509 certificate in bytes.
+ * @param[in,out]  cert_public_key_raw  the public key raw data in cert, only rsa and ecc public key will output
+ * @param[in,out]  public_key_raw_size  The size of the public key raw data in cert
+ * @param[out]     oid                  public key encrypt algo oid
+ * @param[out]     oid_size             the size of public key encrypt algo oid
+ *
+ * @retval  true   Retrieve the public key encrypt algo oid successfully.
+ * @retval  false  Invalid certificate, or the oid_size/public_key_raw_size is too small for the result.
+ **/
+
+bool libspdm_x509_get_public_key_info(const uint8_t *cert, size_t cert_size,
```

Comment:
I am wondering why we need add a new API.

Can we just reuse existing APIs to get the public_key_info in spdm_crypt_lib?

Reply:
I have changed the code and deleted the new API. Thanks.